### ISSUE TITLE
feat(profileReducer): PROFILE_UPDATE_SUCCESS action updates profile state

### ISSUE THREAD

**Issue by serhiipalash** [prescottprue/react-redux-firebase#395]
I see that `PROFILE_UPDATE_SUCCESS` action has `payload` with new profile data, but it doesn't change anything in redux state. Why?

My version is 2.0.0

I think need to add logic for this in `profileReducer`
<img width="1280" alt="screen shot 2018-02-03 at 7 47 45 pm" src="https://user-images.githubusercontent.com/14208343/35769931-95996c7e-091b-11e8-81ba-44851fd74eed.png">

**Comment by serhiipalash** [comment]
To fix this we have to user our custom `userReducer` that looks the same as `profileReducer`, but it updates state when we update user data in database.

**Comment by prescottprue** [comment]
@serhiipalash Yes it does seem like this should be added to the reducer. Open to a PR if you get to it, otherwise it will go on the list for the next release.

**Comment by serhiipalash** [comment]
This works for us 
https://github.com/serhiipalash/react-redux-firebase/commit/d60d5adde9b120fe42f1476ae2e1e1eebbba733b

I can create PR if you give me rights.

**Comment by prescottprue** [comment]
Thanks for the sample, that is nice and simple. A PR would be awesome! You should be able to create a PR without rights to merge or am I missing something?

**Comment by serhiipalash** [comment]
<img width="1280" alt="screen shot 2018-02-09 at 1 10 31 am" src="https://user-images.githubusercontent.com/14208343/36003311-13f72d22-0d36-11e8-94a4-435247f441c4.png">

**Comment by serhiipalash** [comment]
I can't push my working branch without collaborator rights.

**Comment by serhiipalash** [comment]
> This works for us
> serhiipalash/react-redux-firebase@d60d5ad

This if fork ^

**Comment by prescottprue** [comment]
@serhiipalash You don't want to push right to the remote. You want to open a PR from your fork to master.

**Comment by prescottprue** [comment]
I went ahead and did it this time around. When working on big open source projects it is often common to go this route so you don't have to hand out rights to anyone wanting to add to the codebase.

Thanks for the PR (I know I just made it, but it is from your fork :)).

**Comment by serhiipalash** [comment]
Ahaa, I was in the middle of the process with creating my PR :)
Thanks!
Sorry for misunderstanding, I never did PRs from forks before.

**Comment by prescottprue** [comment]
All good. Part of why open source is awesome is all of the free learning!

**Comment by serhiipalash** [comment]
Hi @prescottprue !
Sorry, if this is dump question, but how does `profileDecorator` or `profileFactory` work?

This is my config

```
const rrfConfig = {
  userProfile: 'users',

  profileFactory: (userData, profileData) => {
    console.log('we are in profileFactory')

    const { user } = userData

    return {
      email: user.email,
    }
  },

  profileDecorator: userData => {
    console.log('we are in profileDecorator')

    return { email: userData.email }
  },
}
```

I can't see neither `we are in profileFactory` or `we are in profileDecorator` in logs.

**Comment by prescottprue** [comment]
`profileFactory` is a factory function (creates things) for the profile. That means it is called every time the user profile is being created or updated this factory is called to create the profile (i.e. on `login`, `signup`, and `createUser`).

[Once `v2.0.4`](https://github.com/prescottprue/react-redux-firebase/pull/407) is released `profileFactory` will also fire for Firestore profiles (it was intended to before, but there is a bug within it getting fixed by `v2.0.4`)

You will often get quicker responses to implementation/usage questions [over gitter](https://gitter.im/redux-firestore/Lobby).

**Comment by serhiipalash** [comment]
Thanks!

**Comment by prescottprue** [comment]
Included in [`v2.0.4` release](https://github.com/prescottprue/react-redux-firebase/releases/tag/v2.0.4). Thanks for reporting and providing the PR!

**Comment by serhiipalash** [comment]
> Included in v2.0.4 release. Thanks for reporting and providing the PR!

Great! Thanks for your help!
